sdk/monitor/monitor-query/src/logsQueryClient.ts

```diff
@@ -115,11 +115,23 @@ export class LogsQueryClient {
 
     const parsedBody = JSON.parse(rawResponse.bodyAsText!);
     flatResponse.tables = parsedBody.tables;
-    return {
+    const result: LogsQueryResult = {
       tables: flatResponse.tables.map(convertGeneratedTable),
       statistics: flatResponse.statistics,
-      visualization: flatResponse.render
+      visualization: flatResponse.render,
+      error: flatResponse.error
     };
+    if (result.tables && result.error) {
+      result.logsQueryResultStatus = "Partial";
+    } else if (result.error && !result.tables) {
+      result.logsQueryResultStatus = "Failed";
+    } else {
+      result.logsQueryResultStatus = "Success";
+    }
+    if (options?.throwOnAnyFailure && result.logsQueryResultStatus !== "Success") {
+      throw result;
```

Comment:
hmmm I'm not sure if throwing a result object is the right thing here - we should through something that is an Error object

Reply:
Yeah - we are just following what search documents is doing - in case if i throw an error - then the user won't get the data for queries that succeeded.

Reply:
I looked at the guidelines on [error handling](https://azure.github.io/azure-sdk/typescript_implementation.html#general-error-handling) and while it's not definitively stated that we should always use an ErrorLike I can think of a few reasons why we should be careful with this approach:

1. I expect an error to have a name and message property (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/message) 
2. I am writing a UI app and I just put error.toString() in a banner - what would that look like? [object Object]? 
 - I guess it's better if I use promise API but even then, `catch` expects `rejected` with a reason as the argument...

Not sure, I think it feels weird to throw something that is not an error. @bterlson or @chradek might convince me otherwise and maybe I have it all wrong in which case it'll be a good chance to learn what is right 😄

Reply:
I think I agree we should have a name, code, and message property (i.e. be error-like), otherwise error handling becomes awkward. Additional properties are fine if needed, though should be used sparingly since TS tooling doesn't help with them without the user explicitly casting.

It's unfortunate that search-documents isn't this way, probably in part because thrown errors don't show up in the API review 😕 I wonder what motivated the search-documents design, does anyone know?

Reply:
We'd have to ask @xirzec but I wouldn't be surprised if it was an oversight. It wouldn't have shown up in the API review file and the PR that added it was pretty massive (https://github.com/Azure/azure-sdk-for-js/pull/7482)

I do also think we should throw something error-like. I suspect users would expect message to be available at the very least.

Reply:
I am not sure about the complete reasoning for search-documents, but for monitor-query and search-documents - one thing is true that we have multiple results being returned in that one response. For monitor query, in a single response we have multiple result objects and each of them has an error property that has the structure - similar to the guidelines - https://apiview.dev/Assemblies/Review/06691b2d79d543a58a95ad4d0ea22ed4#@azure/monitor-query!ErrorInfo:interface 

We need to figure out what to do when user sets the "throwOnAnyError" flag -
- Should we throw the first error?
- Should we throw a list of all errors
- In cases of partial errors (i.e, for responses that have both table property and error property), should we throw both table and error property ?
- Other languages (.NET and Java) are throwing the errors on cases of complete errors and are throwing on partial errors if that flag is set.

Reply:
This is what python is doing - https://gist.github.com/rakshith91/d08caf649feac0081703d4c1e5850326